CSharp/ArticleProjects/TaskDependencies/Program.cs

```diff
@@ -58,105 +56,188 @@ public static void Main(string[] args)
+
+                    ODATADetailLevel detail = new ODATADetailLevel()
+                    {
+                        SelectClause = "id, executionInfo"
+                    };
+
+                    foreach (CloudTask task in tasks)
+                    {
+                        // Populate the task's properties with the latest info from the Batch service
+                        await task.RefreshAsync(detail);
+
+                        Console.WriteLine(task.Id);
+
+                        if (task.ExecutionInformation.ExitCode.HasValue)
+                        {
+                            Console.WriteLine("Exit code value: {0}", task.ExecutionInformation.ExitCode.Value);
+                        }
+
+                        if (task.ExecutionInformation.SchedulingError != null)
+                        {
+                            // A scheduling error indicates a problem starting the task on the node. It is important to note that
+                            // the task's state can be "Completed," yet still have encountered a scheduling error.
+
+                            //allTasksSuccessful = false;
+
+                            Console.WriteLine("WARNING: Task [{0}] encountered a scheduling error: {1}", task.Id, task.ExecutionInformation.SchedulingError.Message);
+                        }
+                        else if (task.ExecutionInformation.ExitCode != 0)
+                        {
+                            // A non-zero exit code may indicate that the application executed by the task encountered an error
+                            // during execution. As not every application returns non-zero on failure by default (e.g. robocopy),
+                            // your implementation of error checking may differ from this example.
+
+                            //allTasksSuccessful = false;
+
+                            Console.WriteLine("WARNING: Task [{0}] returned a non-zero exit code - this may indicate task execution or completion failure.", task.Id);
+                        }
+
+                    }
 
-                    Console.WriteLine("All tasks completed successfully.");
-                    Console.WriteLine();
-                }
-                catch (TimeoutException e)
-                {
-                    Console.WriteLine(e);
                 }
+            }
+            finally
+            {
+                BatchClient batchClient = await BatchClient.OpenAsync(cred);
 
                 // Clean up the resources we've created in the Batch account
                 Console.Write("Delete job? [yes] no: ");
                 string response = Console.ReadLine().ToLower();
                 if (response != "n" && response != "no")
                 {
-                    await batchClient.JobOperations.DeleteJobAsync(job.Id);
+                    await batchClient.JobOperations.DeleteJobAsync(jobId);
+                    Console.WriteLine("Job '{0}' deleted.", jobId);
```

Comment:
Okay, good to know. This was in the example already; I just changed it to use the jobId variable and put it in a Finally block. As I said in my email, I think I'm going to need to dig deeper here than I have time for right now, so perhaps we can review the whole sample together to make sure what's there all makes sense.